src/shared/utils/reactProdInvariant.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule reactProdInvariant
+ */
+'use strict';
+
+/**
+ * WARNING: DO NOT manually require this module.
+ * This is a replacement for `invariant(...)` used by the error code system
+ * and will _only_ be required by the corresponding babel pass.
+ * It always throw.
+ */
+function reactProdInvariant(code, a, b, c, d, e, f) {
+  var argCount = arguments.length - 1;
+  var error = new Error('');
+
+  var format = (
+    'React: production error #' + code + '. ' +
+    'Visit http://facebook.github.io/react/docs/' +
+    'error-codes.html?invariant=' + code
+  );
+
+  while (argCount > 0) {
+    format += '&args=%22%s%22';
```

Comment:
Why are we putting everything in quotes? This is a query string so we don't need them (then we also don't have to remove them on the other end). We can also just build up and use `encodeURIComponent` directly here instead of building the %s and replacing immediately (which btw, you aren't current encoding the value of args which we do need to do).

Typically also for query params with the same name to be handled by a server side program you would do `args[]=`. It doesn't really matter since we're doing the parsing on the other end but might be a nice touch in case we did decide to do something different later.